File: web/src/apis/project.ts
import { httpClient } from './request'

export interface Project {
  id: number
  name: string
  desc: string
  created_at: string
  updated_at: string
}

export async function createProject(payload: { name: string; desc?: string }): Promise<string> {
  const res = await httpClient.post<void>('/api/project', payload)
  return res.message ?? ''
}

export async function deleteProject(id: number): Promise<string> {
  const res = await httpClient.delete<void>(`/api/project/${id}`)
  return res.message ?? ''
}

export async function listProjects(): Promise<Project[]> {
  const res = await httpClient.get<Project[]>('/api/project')
  return res.data ?? []
}

export async function getProject(id: number): Promise<Project> {
  const res = await httpClient.get<Project>(`/api/project/${id}`)
  if (!res.data) {
    throw new Error('empty data')
  }
  return res.data
}

export async function updateProject(id: number, payload: { name: string; desc?: string }): Promise<string> {
  const res = await httpClient.put<void>(`/api/project/${id}`, payload)
  return res.message ?? ''
}

File: server/server/http/handler/project.go
package handler

import (
	"net/http"
	"strconv"

	"gitee.com/openeuler/PilotGo-plugin-llmops/server/server/service/project"
	"github.com/gin-gonic/gin"
)

var projectSrv = project.GetProjectService()

type createProjectReq struct {
	Name string `json:"name"`
	Desc string `json:"desc"`
}

func CreateProject(c *gin.Context) {
	var req createProjectReq
	if err := c.ShouldBindJSON(&req); err != nil || req.Name == "" {
		ResponseError(c, http.StatusBadRequest, "invalid request")
		return
	}
	if err := projectSrv.AddProject(req.Name, req.Desc); err != nil {
		ResponseError(c, http.StatusInternalServerError, err.Error())
		return
	}
	ResponseOK(c, "created")
}

type updateProjectReq struct {
	Name string `json:"name"`
	Desc string `json:"desc"`
}

func UpdateProject(c *gin.Context) {
	idStr := c.Param("id")
	id, err := strconv.Atoi(idStr)
	if err != nil || id <= 0 {
		ResponseError(c, http.StatusBadRequest, "invalid id")
		return
	}

	var req updateProjectReq
	if err := c.ShouldBindJSON(&req); err != nil || req.Name == "" {
		ResponseError(c, http.StatusBadRequest, "invalid request")
		return
	}

	if err := projectSrv.UpdateProject(id, req.Name, req.Desc); err != nil {
		ResponseError(c, http.StatusNotFound, err.Error())
		return
	}
	ResponseOK(c, "updated")
}

func DeleteProject(c *gin.Context) {
	idStr := c.Param("id")
	id, err := strconv.Atoi(idStr)
	if err != nil || id <= 0 {
		ResponseError(c, http.StatusBadRequest, "invalid id")
		return
	}
	if err := projectSrv.DeleteProject(id); err != nil {
		ResponseError(c, http.StatusNotFound, err.Error())
		return
	}
	ResponseOK(c, "deleted")
}

func ListProjects(c *gin.Context) {
	projects, err := projectSrv.GetProjectsList()
	if err != nil {
		ResponseError(c, http.StatusInternalServerError, err.Error())
		return
	}
	Response(c, projects)
}

func GetProject(c *gin.Context) {
	idStr := c.Param("id")
	id, err := strconv.Atoi(idStr)
	if err != nil || id <= 0 {
		ResponseError(c, http.StatusBadRequest, "invalid id")
		return
	}
	p, err := projectSrv.GetProjectByID(id)
	if err != nil {
		ResponseError(c, http.StatusNotFound, err.Error())
		return
	}
	Response(c, p)
}
